File: modules/fun/rps.js
exports.init = (bot) => {
  bot.command('rps', (ctx) => {
    const choices = ['rock', 'paper', 'scissors'];
    const args = ctx.message.text.split(' ').slice(1);
    if (!args[0] || !choices.includes(args[0].toLowerCase())) {
      return ctx.reply('Usage: /rps <rock|paper|scissors>');
    }
    const userChoice = args[0].toLowerCase();
    const botChoice = choices[Math.floor(Math.random() * choices.length)];
    let result = '';
    if (userChoice === botChoice) result = "It's a tie!";
    else if (
      (userChoice === 'rock' && botChoice === 'scissors') ||
      (userChoice === 'paper' && botChoice === 'rock') ||
      (userChoice === 'scissors' && botChoice === 'paper')
    ) result = 'You win!';
    else result = 'You lose!';
    ctx.reply(`You chose ${userChoice}, I chose ${botChoice}. ${result}`);
  });
};

exports.help = [
  { name: '/rps', description: 'Play Rock-Paper-Scissors. Usage: /rps <rock|paper|scissors>', category: 'FUN' }
];

File: handlers/fun/rps.py
"""
RPS command - Rock Paper Scissors game
"""
import logging
import random
from telegram import Update
from telegram.ext import ContextTypes

logger = logging.getLogger(__name__)

COMMAND_INFO = {
    "name": "rps",
    "aliases": ["rockpaperscissors"],
    "description": "Play Rock Paper Scissors",
    "usage": "/rps <rock/paper/scissors>",
    "category": "fun"
}

CHOICES = ['rock', 'paper', 'scissors']
EMOJIS = {
    'rock': '🪨',
    'paper': '📄',
    'scissors': '✂️'
}


async def handle(update: Update, context: ContextTypes.DEFAULT_TYPE):
    """Play rock paper scissors"""
    message = update.message
    
    if not context.args:
        await message.reply_html(
            "🎮 <b>Rock Paper Scissors</b>\n\n"
            "<b>Usage:</b> <code>/rps &lt;choice&gt;</code>\n\n"
            "<b>Choices:</b> rock, paper, scissors\n"
            "🪨 Rock beats ✂️ Scissors\n"
            "📄 Paper beats 🪨 Rock\n"
            "✂️ Scissors beats 📄 Paper"
        )
        return
    
    user_choice = context.args[0].lower()
    
    # Validate choice
    if user_choice not in CHOICES:
        await message.reply_text(
            f"❌ Invalid choice! Choose: rock, paper, or scissors"
        )
        return
    
    # Bot chooses
    bot_choice = random.choice(CHOICES)
    
    # Determine winner
    if user_choice == bot_choice:
        result = "🤝 It's a tie!"
        outcome = "tie"
    elif (
        (user_choice == 'rock' and bot_choice == 'scissors') or
        (user_choice == 'paper' and bot_choice == 'rock') or
        (user_choice == 'scissors' and bot_choice == 'paper')
    ):
        result = "🎉 You won!"
        outcome = "win"
    else:
        result = "💔 You lost!"
        outcome = "lose"
    
    # Build message
    msg = f"🎮 <b>Rock Paper Scissors</b>\n\n"
    msg += f"<b>You chose:</b> {EMOJIS[user_choice]} {user_choice.title()}\n"
    msg += f"<b>I chose:</b> {EMOJIS[bot_choice]} {bot_choice.title()}\n\n"
    msg += f"<b>{result}</b>"
    
    await message.reply_html(msg)

